app/views/apipie_dsl/apipie_dsls/help.html.erb

```diff
@@ -1,58 +1,340 @@
+  </tr>
+  <tr>
+    <td><%= _('Variable') %></td>
+    <td><%= _('This input value is fed from parameter values calculated for the host, which the template is being rendered for. This way the template can access a value of a global parameter, which is applied to every host. If that parameter would be overridden e.g. on host subnet level, this value would be fed to this input.') %></td>
+  </tr>
+  <tr>
+    <td><%= _('Puppet parameter') %></td>
+    <td><%= _('Similar to variable input type, however it loads the value for specific smart class parameter. User needs to specify the puppet class and the smart class parameter name when defining the input.') %></td>
+  </tr>
+  </tbody>
+</table>
+
+<p><%= _('All inputs defined for a signle template must have a have unique name, that is then used in ERB code. Inputs can optionally have a description, that is rendered as an inline help, when the template is rendered.') %></p>
+
+<p><%= (_('Once a template input is defined in the Inputs tab and the template is saved, it can be used in the ERB. To load value specified through this input, global macro <code>input</code> is used. The input name must be specified as the argument. E.g. the example below will load the value through input called cpus.').html_safe) %>
+<p>
+
+  <%= apipie_dsl_example("CPUs: " + apipie_erb_wrap("input('cpus')"), "CPUs: 8") %>
+
+<p><%= (_("While rendering the template in preview mode, the input either loads the value or uses a placeholder in case, the input is not ready for the rendering. If the template is rendered for real and the input is not ready, an error is raised. The readyness criteria differs per input type, e.g. for Fact inputs linked to host's fact with name \"cpu_count\" is ready when @host facts contains value for such fact.")) %> </p>
+
+<p><%= _('As it was mentioned above, different template types support different template inputs. See the table below for available combinations.') %></p>
+
+<table class="table table-condensed">
+  <thead>
+  <tr>
+    <th><%= _('Template type') %></th>
+    <th><%= _('Available input types') %></th>
+  </tr>
+  </thead>
+  <tbody>
+  <tr>
+    <td>Provisioning template</td>
+    <td>Fact value, Variable, Puppet parameter</td>
+  </tr>
+  <tr>
+    <td>Partition table</td>
+    <td>Fact value, Variable, Puppet parameter</td>
+  </tr>
+  <tr>
+    <td>Report template</td>
+    <td>User input</td>
+  </tr>
+  <tr>
+    <td>Job template</td>
+    <td>User input, Fact value, Variable, Puppet parameter</td>
+  </tr>
+  </tbody>
+</table>
+
+<h2 id="snippets"><%= _('Snippets') %></h2>
+
+<p><%= _('Snippets are special kind of templates that can be easily reused in multiple other templates. Imagine there is a bootstrap procedure to be run on every provisioned host. At the same time, there are multiple kickstart templates. To avoid duplication a snippet can be used. Assuming there is a snippet created with a name Bootstrap, following example will inject it in a template') %></p>
```

Comment:
` <p><%= _('Snippets are special kind of templates that can be easily reused in multiple other templates. Imagine there is a bootstrap procedure to be run on every provisioned host. At the same time, there are multiple kickstart templates. To avoid duplication a snippet can be used. Assuming there is a snippet created with a name Bootstrap, following example will inject it in a template') %></p>`

To:

` <p><%= _('Snippets are smaller, reusable templates that can be used to customize other templates. For example,  a bootstrap procedure that must run on every provisioned host, but to achieve this requires modifying multiple kickstart templates. To avoid duplication, a snippet can be used. Assuming there is a snippet created with a name Bootstrap, following example will inject it into a template') %></p>`